README.md

```diff
@@ -44,16 +44,16 @@ out-of-tree plugins.
 
 ## Compilation from Source
 
-Check out the project:
+Check out the project in your `$GOPATH` or set `$GOPATH` to be the folder where you want to work in
 
 ~~~
+% [ -z "$GOPATH" ] && export GOPATH=`pwd`
```

Comment:
This won't work on Windows where `export` is not defined. Also, this is going to create a bunch of folders under the current directory, when they might've wanted to just checkout in this directory.

Instead maybe a block after this that says "If you get `cannot find package` errors when running `make`, ensure that your checked out copy of coredns is available at `$GOPATH/src/github.com/coredns`. You may need to set `$GOPATH` if its not set already. See https://github.com/golang/go/wiki/GOPATH for more information.".

Reply:
Trying to get this moving (one way or another):
* The new instructions may not work for windows, but then the current ones will not neither. Having not used windows since Windows XP, I will let someone else help them !
* Telling Unix users who are unfamiliar something about `$GOPATH` may point them in the right direction but does not provide the steps required before running `make` to "yield a `coredns` binary"
* For people attempting to use `$CWD`, if the user is savvy enough to have cloned CoreDNS within `$GOPATH/src/coredns` folder, I believe the current instructions will be understood (as they likely needed no instructions in the first place)

So unless some better step-by-step instructions can be found, or some change made to help windows users (which I can not provide having not used Windows since XP), I still believe the current patch is a step in the right direction and is worth merging.

Reply:
Maybe just use `$GOPATH` and refer to some official GOPATH docs from the Go project otherwise. Setting GOPATH to pwd is a bit of a no-no, so don't do that.